internal/storage/postgres/tree/get_test.go

```diff
@@ -0,0 +1,662 @@
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, sensor)
+	})
+}
+
+func TestTreeRepository_GetTreeClusterByTreeID(t *testing.T) {
+	t.Run("should return tree cluster for the given tree ID", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		suite.InsertSeed(t, "internal/storage/postgres/seed/test/tree")
+		r := NewTreeRepository(suite.Store, mappers)
+		treeID := int32(1)
+
+		// when
+		treeCluster, err := r.GetTreeClusterByTreeID(context.Background(), treeID)
+
+		// then
+		assert.NoError(t, err)
+		assert.NotNil(t, treeCluster, "TreeCluster should not be nil")
+		assert.NotZero(t, treeCluster.ID, "TreeCluster ID should not be zero")
+	})
+
+	t.Run("should return ErrTreeClusterNotFound when no tree cluster is linked to the tree", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		r := NewTreeRepository(suite.Store, mappers)
+		treeID := int32(4)
+
+		// when
+		treeCluster, err := r.GetTreeClusterByTreeID(context.Background(), treeID)
+
+		// then
+		assert.ErrorIs(t, err, storage.ErrTreeClusterNotFound, "Expected ErrTreeClusterNotFound error")
+		assert.Nil(t, treeCluster, "TreeCluster should be nil when not found")
+	})
+
+	t.Run("should return error when tree ID does not exist", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		r := NewTreeRepository(suite.Store, mappers)
+		invalidTreeID := int32(999)
+
+		// when
+		treeCluster, err := r.GetTreeClusterByTreeID(context.Background(), invalidTreeID)
+
+		// then
+		assert.Error(t, err, "Expected an error when the tree ID does not exist")
+		assert.Nil(t, treeCluster, "TreeCluster should be nil when tree ID does not exist")
+	})
+
+	t.Run("should return error when tree ID is zero", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		r := NewTreeRepository(suite.Store, mappers)
+		invalidTreeID := int32(0)
+
+		// when
+		treeCluster, err := r.GetTreeClusterByTreeID(context.Background(), invalidTreeID)
+
+		// then
+		assert.Error(t, err, "Expected an error when the tree ID is zero")
+		assert.Nil(t, treeCluster, "TreeCluster should be nil when tree ID is zero")
+	})
+
+	t.Run("should return error when tree ID is negative", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		r := NewTreeRepository(suite.Store, mappers)
+		invalidTreeID := int32(-1)
+
+		// when
+		treeCluster, err := r.GetTreeClusterByTreeID(context.Background(), invalidTreeID)
+
+		// then
+		assert.Error(t, err, "Expected an error when the tree ID is negative")
+		assert.Nil(t, treeCluster, "TreeCluster should be nil when tree ID is negative")
+	})
+
+	t.Run("should return error when context is canceled", func(t *testing.T) {
+		// given
+		r := NewTreeRepository(suite.Store, mappers)
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		// when
+		treeCluster, err := r.GetTreeClusterByTreeID(ctx, 1)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, treeCluster)
+	})
+}
+
+func assertExpectedEqualToTree(t *testing.T, expectedTree, tree *entities.Tree) {
```

Comment:
What about checking the `sensorID` and `treeclusterID` here?

Reply:
I test it before calling the function because I need to first get the objects from the database.

```go
		assert.NoError(t, err)
		assert.NotNil(t, tree)
		assert.Equal(t, tree.TreeCluster.ID, treeCluster.ID)
		assert.Equal(t, tree.Sensor.ID, sensor.ID)
		assert.NotNil(t, tree.Sensor)
		assertExpectedEqualToTree(t, tree, testTrees[0])
```
		
please check  `TestTreeRepository_GetByID`

Reply:
I would like to have everything in one function, but it's okay 👍 I can live with it.